Automattic/simplenote-ios
# Migrate CoreData to shared app group directory

This PR is a continuation of #1329 I have cherry picked the commits from the feature/widgets-shared-db branch to a new branch that is separate form the widgets and only focuses on the migration of the database.

### Fix
This PR makes changes to the CoreData database for Simplenote iOS.  It approaches two changes, first it moves the CoreData stack into its own class, and second it migrates the data from the in app storage to a shared directory that can be accessed by other SNiOS extensions.


### Test
To test this PR you need to have a simulator or device setup with a database that was configured in the previous location first.  So..
1. Load the develop branch on your device/sim and log into simplenote
2. Stop operation of the app on that device
3. Change branches to feature/shared-db-migration
4. Compile this branch to your device.
5. When the app loads the migration should happen automatically.

In the app if the migration succeeds or fails it shouldn't be visible to the user as if it fails then the app should default to using the previous database location.  To tell if this has worked, check the console logs, there should be prints from the app that the database was migrated successfully.
If you run the app again after a successful migration you should see a message "Core Data Migration not required" and the app should load normally

### Review

> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
